airgun/views/computeresource.py

```diff
@@ -0,0 +1,50 @@
+from widgetastic.widget import Checkbox, Select, Text, TextInput, Widget
+from airgun.views.common import BaseLoggedInView, SearchableViewMixin
+from airgun.widgets import FilteredDropdown
+
+
+class ClickOnArrowButton(Widget):
+    ROOT = "/html/body/div[2]/div/div/table/tbody/tr[1]/td[3]/div"
+    open_dropdown = Text(".//a[contains(@data-toggle, 'dropdown')]")
+    delete = Text(
+        ".//li[not(contains(@style, 'display: none'))][contains(.,'Delete')]")
+
+    def fill(self, value):
+        self.open_dropdown.click()
+        self.delete.click(handle_alert=True)
+
+
+class ComputeResourcesView(BaseLoggedInView, SearchableViewMixin):
+    title = Text("//h1[text()='Compute Resources']")
+    new = Text("//a[contains(@href, '/compute_resources/new')]")
+    edit = Text("//*[text()='Edit']")
+    action_list = ClickOnArrowButton()
+
+    @property
+    def is_displayed(self):
+        """Check if the right page is displayed"""
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+
+class ResourceProviderDetailsView(BaseLoggedInView):
+    name = TextInput(id='compute_resource_name')
+    provider = FilteredDropdown(id='s2id_compute_resource_provider')
+    description = TextInput(id='compute_resource_description')
+    url = TextInput(id='compute_resource_url')
+    display_type = Select(id='compute_resource_display_type')
+    console_passwords = Checkbox(id='compute_resource_set_console_password')
+    user = TextInput(id='compute_resource_user')
+    password = TextInput(id='compute_resource_password')
+    vcenter = TextInput(id='compute_resource_server')
+    datacenter = Text("//*[@class='caption']//*[text()='Load Datacenters']")
```

Comment:
Bad name for variable, should have been something like load_datacenters because it's for the button Load Datacenters but I never needed it anyway (to click on that button).

Reply:
Hmm, I still think to distinguish each provider type into separate nested views.
```
+    name = TextInput(id='compute_resource_name')
+    provider = FilteredDropdown(id='s2id_compute_resource_provider')
+    description = TextInput(id='compute_resource_description')
```
and
```
    @View.nested
    class docker_provider(View):
        url = TextInput(id='compute_resource_url')
        user = TextInput(id='compute_resource_user')
        password = TextInput(id='compute_resource_password')
        email = TextInput(id='...')
and etc
```

Reply:
@abalakh thoughts?

Reply:
No preference, both solutions are temporary until we find some time to learn conditional views.

Reply:
Okay, so now we actually have implementation for conditional views. Can you look whether it can be helpful for you?

Examples:
https://github.com/SatelliteQE/airgun/blob/master/airgun/views/product.py#L150
https://github.com/SatelliteQE/airgun/blob/master/airgun/views/partitiontable.py#L48

Reply:
I think it's not really necessary to use conditional views for every provider and I'd rather have it as it is. But If I really need to use conditional views what's the best approach? I don't know If I can use SatTab or anything else from views/common for every provider?